ako-gateway-api/k8s/ako_init.go

```diff
@@ -224,6 +228,31 @@ func (c *GatewayController) addIndexers() {
 				}
 				return []string{}, nil
 			},
+			akogatewayapilib.AviInfraSettingGatewayClassIndex: func(obj interface{}) ([]string, error) {
+				gwClass, ok := obj.(*gatewayv1.GatewayClass)
+				if !ok {
+					return []string{}, nil
+				}
+				if gwClass.Spec.ParametersRef != nil {
+					infraSetting := string(gwClass.Spec.ParametersRef.Group) + "/" + string(gwClass.Spec.ParametersRef.Kind) + "/" + gwClass.Spec.ParametersRef.Name
+					return []string{infraSetting}, nil
```

Comment:
Is this code needed ?

Shouldn't it be removed as per our discussion with Anurag